docs/shrc_diag_util.md

```diff
@@ -22,104 +22,124 @@
-    - `CLASSPATH` : the type of data that was written.
+If you are using a custom class loader, class path entries in the `SharedClassURLClasspathHelper` must be *confirmed* before classes can be found for these entries. More information about confirmed entries is available in the `SharedClassHelper` [API documentation](api-overview.md#monitoring-and-management).
 
-### ROMClasses
+The most common cause for classes not being stored in the cache is due to space constraints. Make sure that you set an appropriate size for the cache, as described in [Creating a shared classes cache](shrc.md#creating-a-shared-classes-cache). Setting a soft maximum size is recommended, because you can adjust the soft maximum size that is set for the cache after it is created. See [Setting a soft maximum size](xscmx.md#setting-a-soft-maximum-size).
 
-: This example shows an entry for a single `ROMClass`:
+Storing classes can also be a problem if the cache is opened read-only or if the class does not exist on the file system because it is sourced from a URL location.
 
-        1: 0x2234F7DC ROMCLASS: java/lang/Runnable at 0x213684A8
-                Index 1 in class path 0x2234FA6C
+If you are instrumenting bytecode by using a JVMTI agent or `java.lang.instrument` agent, the following rules apply:
 
-    - `1` : the ID of the VM that wrote this data.
-    - `0x2234F7DC` : the address where the metadata about the class is stored.
-    - `ROMCLASS` : the type of data that was stored.
-    - `java/lang/Runnable` : the name of the class.
-    - `0x213684A8` : the address where the class was stored.
-    - `Index 1` : the index in the class path where the class was loaded from.
-    - `0x2234FA6C` : the address of the class path against which this class is stored.
+- Redefined classes are never stored in the cache.
+- Retransformed classes are stored only if you specify the `-Xshareclasses:cacheRetransformed` suboption when you start your application.
 
-: Stale classes are marked with `!STALE!`. Any partition or modification context that is used when the class is stored is also shown.
+If a running application uses its own class loader and you are using a `SecurityManager`, you must grant the class loader permission to `SharedCachePermission` before they can share classes. For more information, see [Support for custom class loaders](shrc.md#support-for-custom-class-loaders).
 
-### AOT methods
+In very rare cases, problems with finding or storing classes might be due to cache corruption. If the VM detects that a cache is corrupt, it attempts to destroy the cache and re-create it. If the VM cannot re-create the cache, it starts only if the `-Xshareclasses:nonfatal` suboption is specified on the command line, but without using the shared cache. Try using the `-Xshareclasses:destroy` cache utility to remove the specific cache and re-create it. You might need to specify the `cacheDir` and `name` suboptions if the cache is not using the default settings.
```

Comment:
> "You might need to specify the `cacheDir` and `name` suboptions if the cache..."

These options should have `=` in them. As in other places talking about these options, we are using `name=<cache_name>` and `cacheDir=<directory>`. better to be consistent.